libs/assigner_runner/src/block_parser.cpp

```diff
@@ -0,0 +1,89 @@
+#include "zkevm_framework/assigner_runner/block_parser.hpp"
+
+#include <boost/algorithm/hex.hpp>
+#include <boost/endian.hpp>
+#include <boost/endian/conversion.hpp>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+
+#include "block_schema.def"
+#include "zkevm_framework/json_helpers/json_helpers.hpp"
+
+template<typename T>
+std::optional<std::string> to_std_bytes(const boost::json::value &json_value, T &dst) noexcept {
+    std::vector<uint8_t> bytes;
+    json_helpers::to_bytes(json_value, bytes);
+    for (std::size_t i = 0; i < bytes.size(); i++) {
+        dst[i] = std::byte{bytes[i]};
+    }
+    return std::nullopt;
+}
+
+static std::optional<std::string> handle_message(const boost::json::object &val,
```

Comment:
Seems we can use `void` as return type since no errors are returned

Reply:
the unction can return error now